src/material-experimental/mdc-form-field/_form-field-high-contrast.scss

```diff
@@ -0,0 +1,36 @@
+@use '../../cdk/a11y';
+
+@mixin private-form-field-high-contrast() {
+  $focus-indicator-width: 3px;
+  $focus-indicator-style: dashed;
+
+  // The outline of the `fill` appearance is achieved through a background color
+  // which won't be visible in high contrast mode. Add an outline to replace it.
+  .mat-form-field-appearance-fill .mat-mdc-text-field-wrapper {
+    @include a11y.high-contrast(active, off) {
+      outline: solid 1px;
+    }
+  }
+
+  // If a form field with fill appearance is focused, update the outline to be
+  // dashed and thicker to indicate focus.
+  .mat-form-field-appearance-fill.mat-focused .mat-mdc-text-field-wrapper {
+    @include a11y.high-contrast(active, off) {
+      outline: $focus-indicator-style $focus-indicator-width;
+    }
+  }
+
+  // For form fields with outline appearance, we unset the outline solid borders
+  // and show a dashed thick border to indicate focus.
+  .mat-mdc-form-field.mat-focused .mdc-notched-outline {
+    @include a11y.high-contrast(active, off) {
+      .mdc-notched-outline__leading,
+      .mdc-notched-outline__trailing,
+      .mdc-notched-outline__notch, {
+        border: none;
+      }
+
+      border: $focus-indicator-style $focus-indicator-width;
```

Comment:
Could we just make it thicker by adding the outline on top of the border? I'm a little hesitant with changing properties that affect layout only for high contrast mode. Same goes for the identical changes below.

Reply:
I've considered it initially; but it didn't look to good; but I see the argument of not changing too much of the layout (even though some other code in the repo does it). Changed.

Reply:
Also updated the screenshot above in case you want to see how it looks.